scripts/train_lm.py

```diff
@@ -146,12 +146,39 @@ def WriteMetaparameters(metaparameters, ngram_order, num_train_sets, out_file):
 
 def FormatMetaparameters(metaparameters):
     out = []
+    # get the maximum length of decimals of metaparameters
+    decimal_length = 0
     for param in metaparameters:
-        x = '{:.3f}'.format(param)
-        if x == '0.00':
-            x = '{:.3g}'.format(param)
-        out.append(x)
-
+        temp_length = len(str(param))
+        if decimal_length < temp_length:
+            decimal_length = temp_length
+    decimal_length -= 2
+    
+    temp_original = metaparameters[0]
+    temp = round(temp_original, 3)
+    for param in metaparameters[1:]:
+        round_index = 3
+        x_original = param
+        x = round(x_original, round_index)
+        while(temp == x and round_index <= decimal_length):
```

Comment:
Sorry I just saw your comments. 
As for the first one, I see your point. It seems that when the number is too close to 1, the input to RoundToProperDigit() is 1.0 and this function returns 1.0, which cause a false alarm. 
While I think when this indeed happens, we should pay attention to this number since it's too close to 1. But I'm not fully sure. 
As for the second point, I think if we choose a certain number as precision, for example 3, it doesn't matter much whether "0.000001" or "0.000001234567", right?